packages/types/src/context.ts

```diff
@@ -1,2 +1,72 @@
 export type Context = Record<string, unknown>;
-export type Contexts = Record<string, Context>;
+
+export interface Contexts extends Record<string, Context | undefined> {
```

Comment:
Why the union with `undefined` here?

Reply:
It's required for the `app`, `device`, etc properties below to be optional.